docs/yul.rst

```diff
@@ -9,18 +9,19 @@ Yul
 Yul (previously also called JULIA or IULIA) is an intermediate language that can
 compile to various different backends
 (EVM 1.0, EVM 1.5 and eWASM are planned).
-Because of that, it is designed to be a usable common denominator of all three
+Because of this, it is designed to be a usable common denominator of all three
 platforms.
 It can already be used for "inline assembly" inside Solidity and
-future versions of the Solidity compiler will even use Yul as intermediate
+future versions of the Solidity compiler will use Yul as intermediate
 language. It should also be easy to build high-level optimizer stages for Yul.
 
-.. note::
+Yul is implemented when using the compiler in strict assembly mode, which you can use
+by adding the ``--strict-assembly`` and ``--optimize`` arguments when using ``solc``.
+
+.. warning::
 
-    Note that the flavour used for "inline assembly" does not have types
-    (everything is ``u256``) and the built-in functions are identical
-    to the EVM opcodes. Please resort to the inline assembly documentation
-    for details.
+    Yul is in active development, and as such does not yet support types (everything is ``u256``),
+    and the built-in functions are identical to the EVM opcodes. Refer to the :ref:`inline assembly documentation <inline-assembly>` for details.
```

Comment:
Yul is indeed in active development, however it does support types. I suggest retaining the note, and adding a sentence to the effect of "Yul is not supported yet because it is under active development."

Reply:
OK, this I am confused by, as the existing text said it doesn't support types and I'm sure that's what @chriseth also said as a reason to create the issue. So I need a little clarity there :)